reset-pass/account-validator-impl/src/java/org/sakaiproject/accountvalidator/logic/impl/ValidationLogicImpl.java

```diff
@@ -387,109 +403,42 @@ public ValidationAccount createValidationAccount(String userRef,
-			userEid = u.getEid();
-			//information about the user that added them
-			User added = u.getCreatedBy();
-			replacementValues.put("addedBy", added.getDisplayName());
-			replacementValues.put("addedByEmail", added.getEmail());
-			
-		} catch (UserNotDefinedException e) {
-			// TODO Auto-generated catch block
-			e.printStackTrace();
-		}
-		
-		
-		
-		
-		//information about the site(s) they have been added to
-		Set<String> groups = authzGroupService.getAuthzGroupsIsAllowed(userId, SiteService.SITE_VISIT, null);
-		log.info("got a list of: " + groups.size());
-		Iterator<String> itg = groups.iterator();
-		StringBuilder sb = new StringBuilder();
-		int siteCount = 0;
-		while (itg.hasNext()) {
-			String groupRef = itg.next();
-			String siteId = developerHelperService.getLocationIdFromRef(groupRef);
-			try {
-				Site s = siteService.getSite(siteId);
-				if (siteCount > 0) {
-					sb.append(", ");
-				}
-				log.info("adding site: " + s.getTitle());
-				sb.append(s.getTitle());
-				siteCount++;
-			} catch (IdUnusedException e) {
-				// TODO Auto-generated catch block
-				e.printStackTrace();
-			}
-			
-		}
-		replacementValues.put("memberSites", sb.toString());
-		replacementValues.put("displayName", userDisplayName);
-		replacementValues.put("userEid", userEid);
-		replacementValues.put("support.email", serverConfigurationService.getString("setup.request", "no-reply@"+serverConfigurationService.getServerName()));
-		replacementValues.put("institution", serverConfigurationService.getString("ui.institution"));
-		String templateKey = getTemplateKey(accountStatus);
-
-		
-		
-		emailTemplateService.sendRenderedMessages(templateKey , userReferences, replacementValues, serverConfigurationService.getString("setup.request", "no-reply@"+serverConfigurationService.getServerName()), serverConfigurationService.getString("setup.request", "no-reply@"+serverConfigurationService.getServerName()));
-		v.setValidationSent(new Date());
-		v.setStatus(ValidationAccount.STATUS_SENT);
-		
 		/*if (ValidationAccount.ACCOUNT_STATUS_PASSWORD_RESET == accountStatus.intValue()) {
 			//A password reset doesn't invalidate confirmation
 			v.setStatus(ValidationAccount.STATUS_CONFIRMED);
 		} else { 
 			v.setStatus(ValidationAccount.STATUS_SENT);
 		}*/
-		v.setFirstName(userFirstName);
-		v.setSurname(userLastName);
-		
-		dao.save(v);
+		v = saveValidationAccount(v);
 		return v;
 	}
+	//Set other details for ValidationAccount and save
+	private ValidationAccount saveValidationAccount(ValidationAccount account){
+		account.setValidationSent(new Date());
+		account.setStatus(ValidationAccount.STATUS_SENT);
+		String userId = EntityReference.getIdFromRef(account.getUserId());
+		try{
+			User u = userDirectoryService.getUser(userId);
+			if (u.getFirstName() != null)
+				account.setFirstName(u.getFirstName());
+
+			if (u.getLastName() != null)
+				account.setSurname(u.getLastName());
+		}
+		catch(UserNotDefinedException e){
+			e.printStackTrace();
```

Comment:
This should be logged with the stacktrace as the throwable, not a raw stacktrace.